core/src/main/java/org/apache/spark/memory/TaskMemoryManager.java

```
@@ -421,6 +421,72 @@ public void freePage(MemoryBlock page, MemoryConsumer consumer) {
     }
   }
 
+  /**
+   * @param size
+   * @return
+   */
+  public MemoryBlock allocatePMemPage(long size) {
+    if (size > MAXIMUM_PAGE_SIZE_BYTES) {
+      throw new TooLargePageException(size);
+    }
+    final int pageNumber;
+    synchronized (this) {
+      pageNumber = allocatedPages.nextClearBit(0);
+      if (pageNumber >= PAGE_TABLE_SIZE) {
+        releaseExtendedMemory(size);
+        throw new IllegalStateException(
```

Do we need to throw an exception or handle it (e.g. fallback or double size with warnning log)? Please address it in another PR.

---

This is same as vanilla spark memory allocate.